This patch is required to fix a segfault on startup in <=0.22.1 if star catalogue files are
added or removed (USE=stars)

Refer: https://github.com/Stellarium/stellarium/issues/2404

This patch may be safely removed for 0.22.2 - it has been patched upstream.

--- a/src/core/modules/StarMgr.cpp
+++ b/src/core/modules/StarMgr.cpp
@@ -592,13 +592,13 @@ void StarMgr::setCheckFlag(const QString& catId, bool b)
 			return;
 		m["checked"]=b;
 		catalogsDescription[idx-1]=m;
-		starSettings["catalogs"]=catalogsDescription;
-		QFile tmp(starConfigFileFullPath);
-		if(tmp.open(QIODevice::WriteOnly))
-		{
-			StelJsonParser::write(starSettings, &tmp);
-			tmp.close();
-		}
+	}
+	starSettings["catalogs"]=catalogsDescription;
+	QFile tmp(starConfigFileFullPath);
+	if(tmp.open(QIODevice::WriteOnly))
+	{
+		StelJsonParser::write(starSettings, &tmp);
+		tmp.close();
 	}
 }
 
